lms/djangoapps/lti_provider/views.py

```diff
@@ -64,8 +70,18 @@ def lti_launch(request, course_id, usage_id):
     # Store the course, and usage ID in the session to prevent privilege
     # escalation if a staff member in one course tries to access material in
     # another.
-    params['course_id'] = course_id
-    params['usage_id'] = usage_id
+    try:
+        course_key, usage_key = parse_course_and_usage_keys(course_id, usage_id)
+    except InvalidKeyError:
+        log.error(
+            'Invalid course key %s or usage key %s from request %s',
+            course_id,
+            usage_id,
+            request
```

Comment:
It dumps the full content of the request, which includes the LTI content and the referrer. My thinking was that this should help an admin figure out who's making requests for a bad item so that it's easier to fix.